### DISCUSSION TITLE
Setting up crowd-sourced translation platform.

### DISCUSSION THREAD

**Discussion by vsrisuknimit** [bafolts/terraforming-mars#1360] [General]
We should have a platform for crowd-sourcing translations.

@alrusdi suggested Pontoon https://github.com/mozilla/pontoon.

Weblate is another alternative. https://weblate.org/

Poeditor is free with Open Source Intiative license.

**Comment by alrusdi** [comment]
Weblate is kinda 17€/mo
I can't manage Pontoon to work on localhost. it is installable on heroku, but requires paid account

**Reply by comradekingu** [reply]
Weblate can be self-hosted.
https://weblate.bubu1.eu/ and https://hosted.weblate.org/hosting/ doesn't charge for libre software projects.

**Reply by alrusdi** [reply]
Please see my comments about Weblate below. It is great but don't meets our requirements

**Comment by vsrisuknimit** [comment]
Weblate can be self-hosted, right? I am not sure how that will work.

Should we ask @bafolts to set up Pontoon since the demo server is already running on paid heroku?

**Comment by alrusdi** [comment]
i don't see self hosted variant of weblate. Can you give me a hint?
Oh... Found it! Nvm https://github.com/WeblateOrg/weblate

**Comment by alrusdi** [comment]
I give up on it. Weblate does not provide all the features we need:
1) Can't edit once created translation key
2) Does not provide a simple way to export en_string => loc_string. It breaks the original strings by sentences

**Comment by vsrisuknimit** [comment]
@alrusdi back to Pontoon then?

**Comment by alrusdi** [comment]
> Pontoon then?

 Only if somebody other than me will do it. I can't figure out of how to deploy it to VDS

**Comment by Heziode** [comment]
@alrusdi I use Weblate to translate a web-app (by the community) for years. My web-app is a Nuxt (that uses nuxt-i18n, so vue-i18n).

The only boring thing is that Vue-i18n uses its own locale file format. To work with Weblate, I use PO files, that is converted into vue-i18n files. This allows supporting every language (even if with their complexity, like multiple plural of Russian, the amazing gender of the French language, etc.)

**Reply by alrusdi** [reply]
Is it possible in Weblate to edit the translation key if it is already saved? Can it export translations as simple map {"string to translate": "translation of the string"?

**Reply by Heziode** [reply]
> Is it possible in Weblate to edit the translation key if it is already saved? 

In case of bilingual files (your case), there is no way to edit the "Source String" in Weblate.

> Can it export translations as simple map {"string to translate": "translation of the string"?

Yep, it supports JSON export: https://docs.weblate.org/en/latest/formats.html#json-files

I would nevertheless like to point out that the use of a string in English (or whatever the language) as a key is a very bad idea.
Every time I have seen programs/site doing this, they have been annoyed to updates the source string.

Using an Id is the best way IMHO (more easy to manage plural, fix typo, etc. using an Id as key, instead of a sentence).

**Comment by kberg** [comment]
Related: #5338
